src/include/migraphx/builtin.hpp

```diff
@@ -63,6 +63,20 @@ struct param
     }
 };
 
+struct add_return
+{
+    std::vector<std::string> output_names;
```

Comment:
Why do we store names?

Reply:
Two reasons: The main reason is this is required to support onnxruntime. We need to use input and output name to match our program inputs/outputs to that in the onnxruntime, so we can put correct buffer in our parameters. Another reason is by storing the output names, we can verify our program outputs match that in onnx files.  The output names can be empty here. If empty, we use output_1, 2,3... as the output names.

Reply:
Why can't they match based on the index?

Reply:
Before calling prog.eval(), we need to set arguments retrieved by calling get_param_shapes(). The argument map includes both inputs and output and there is no guarantee about the order of the arguments (inputs are before outputs, and output parameter are in the same order as that in onnx file).  So how can find which parameter here is which program output?

Reply:
if we really want to use index to match, we have to allocate parameter buffer for each output parameter, then call prog.eval() to compute the result. Then call hipMemcpy to copy result from our allocated buffer to the tensor buffer in onnxruntime, but this introduce additional memory copy, which is not necessary. 

The approach I am using now is using the onnxruntime tensor buffer as the buffer in our argument, so after prog.eval(), the prog output is already in the onnxruntime tensor buffer.

Reply:
removed the output name